test/src/unittests/rhythm/test_superfluxextractor.py

```diff
@@ -17,12 +17,60 @@
-
+from essentia.standard import MonoLoader, SuperFluxExtractor
+from numpy import *
 
 class TestSuperFluxExtractor(TestCase):
 
+
+    def testInvalidParam(self):
+        # All the parameters ranges are theoretically from 0 to infinity
+        # Hence, we use neg. value for invalid checks.
+        self.assertConfigureFails(SuperFluxExtractor(), { 'combine': -1})
+        self.assertConfigureFails(SuperFluxExtractor(), { 'frameSize': -1})
+        self.assertConfigureFails(SuperFluxExtractor(), { 'hopSize': -1})
+        self.assertConfigureFails(SuperFluxExtractor(), { 'ratioThreshold': -1})
+        self.assertConfigureFails(SuperFluxExtractor(), { 'sampleRate': -1})
+        self.assertConfigureFails(SuperFluxExtractor(), { 'threshold': -1})                  
+
+    def testRegressionDubstep(self):
+        audio = MonoLoader(filename=join(testdata.audio_dir, 'recorded', 'dubstep.wav'))()
+      
+        # This test case will use the documented default parameters from recording dubstep.wav
+        onsets = SuperFluxExtractor(combine=30,frameSize=2048,hopSize=256,ratioThreshold=16,
+            sampleRate=44100,threshold=0.5)(audio)
+        
+        # This commented out code was used to obtain reference samples for storing in a file.
+        # save('superfluxdub', onsets)        
+        
+        # Reference samples are loaded as expected values
+        expected_superflux = load(join(filedir(), 'superflux/superfluxdub.npy'))
+        self.assertAlmostEqualVector(onsets, expected_superflux, 1e-5)
+
+    def testRegressionTechnoloop(self):
+        audio = MonoLoader(filename=join(testdata.audio_dir, 'recorded', 'techno_loop.wav'))()
+       
+        # This test case will use peak parameters slighlt ifferent from default from recording techno_loop.wav
+        onsets = SuperFluxExtractor(combine=20,frameSize=2048,hopSize=256,ratioThreshold=8,
+            sampleRate=44100,threshold=0.25)(audio)
+        
+        # This commented out code was used to obtain reference samples for storing in a file.
+        # save('superfluxtechno', onsets)        
+        
+        # Reference samples are loaded as expected values
+        expected_superflux = load(join(filedir(), 'superflux/superfluxtechno.npy'))
+        self.assertAlmostEqualVector(onsets, expected_superflux, 1e-5)
+
+    def _assertVectorWithinVector(self, found, expected, precision=1e-5):
+        for i in range(len(found)):
+            for j in range(1,len(expected)):
+                if found[i] <= expected[j] and found[i] >= expected[j-1]:
+                    if fabs(found[i] - expected[j-1]) < fabs(expected[j] - found[i]):
+                        self.assertAlmostEqual(found[i], expected[j-1], precision)
+                    else:
+                        self.assertAlmostEqual(found[i], expected[j], precision)
+
     def testSilence(self):
         # zeros should return no onsets (empty array)
         self.assertEqualMatrix(SuperFluxExtractor()(zeros(44100)), [])
```

Comment:
The output is 1-D vector, so we can use `assertEqualVector` instead.

Reply:
self.assertEqualMatrix should use self.assertEqualVector